File: src/main/java/com/shadowbring/todo/domain/ToDoUser.java
package com.shadowbring.todo.domain;

import lombok.Data;
import lombok.EqualsAndHashCode;

import javax.persistence.*;
import java.util.List;

@Entity
@Data
@EqualsAndHashCode(exclude = "id")
@SequenceGenerator(name = "userSequence", sequenceName = "userSequence", allocationSize = 1)
public class ToDoUser {

  @Id
  @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "userSequence")
  private Long id;

  @Column(nullable = false)
  private String name;

  @OneToMany(mappedBy = "toDoUser")
  private List<ToDoItem> toDoItems;
}

File: src/main/java/com/shadowbring/todo/domain/ToDoUser.kt
package com.shadowbring.todo.domain

import java.util.*
import javax.persistence.*

@Entity
@SequenceGenerator(name = "userSequence", sequenceName = "userSequence", allocationSize = 1)
data class ToDoUser(

        @Id
        @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "userSequence")
        private var id: Long = 0,

        @Column(nullable = false)
        private var name: String = "",

        @OneToMany(mappedBy = "toDoUser")
        private var toDoItems: List<ToDoItem> = ArrayList()
)